spacemacs/packages.el

```diff
@@ -2381,13 +2381,13 @@ Put (global-hungry-delete-mode) in dotspacemacs/config to enable by default."
     :init
     (progn
       (defun spacemacs/load-yasnippet ()
-          (if (not (boundp 'yas-minor-mode))
-              (progn
-                (let* ((dir (configuration-layer/get-layer-property 'spacemacs :ext-dir))
-                       (private-yas-dir (concat configuration-layer-private-directory "snippets"))
-                       (yas-dir (concat dir "yasnippet-snippets")))
-                  (setq yas-snippet-dirs (list  private-yas-dir yas-dir))
-                  (yas-global-mode 1)))))
+        (if (not (boundp 'yas-minor-mode))
+            (progn
+              (let* ((dir (configuration-layer/get-layer-property 'spacemacs :ext-dir))
+                     (private-yas-dir (concat configuration-layer-private-directory "snippets"))
+                     (yas-dir (concat dir "yasnippet-snippets")))
+                (custom-set-variables '(yas-snippet-dirs (list  private-yas-dir yas-dir)))
```

Comment:
Using `custom-set-variables` will write the symbols and their values at the end of `.spacemacs` at the next dump of the customized variables.
To prevent this we should use `setq`, `push` etc.. instead. It may be required to test if the symbol is bound first. If the symbol is not bound then use `setq` otherwise `push` the 2 default values on top of the user defined values.